### PR TITLE
`useVideoTexture` enhancements

### PR DESCRIPTION
**Pull request by abernier**
### Checklist

- [x] Documentation updated ([example](https://github.com/pmndrs/drei/blob/master/README.md#example))
- [x] Storybook entry added ([example](https://github.com/pmndrs/drei/blob/master/.storybook/stories/Example.stories.tsx))
- [x] Ready to be merged

### CODE REVIEW

**Comment by netgfx** [comment]
@abernier Hello, is it possible to also showcase the HLS capabilities as well? I have an example on codesanbox that showcases it https://codesandbox.io/p/sandbox/drei-hls-usevideotexture-4syd5j

File: .storybook/stories/VideoTexture.stories.tsx [hunk 1]
```diff
@@ -98,3 +98,25 @@ export const UseVideoTextureSceneSt3 = {
...
+          {(texture) => <meshBasicMaterial side={THREE.DoubleSide} map={texture} toneMapped={false} />}
+        </VideoTexture>
+      </Plane>
+    </>
+  )
+}
+
+export const VideoTextureSt4 = {
+  args: {
+    src: 'https://test-streams.mux.dev/x36xhzz/x36xhzz.m3u8', // m3u8 file from: https://hlsjs.video-dev.org/demo/
```
Comment (abernier) [inline .storybook/stories/VideoTexture.stories.tsx:118]: added a story with HLS

File: docs/loaders/video-texture-use-video-texture.mdx
Comment (abernier) [inline docs/loaders/video-texture-use-video-texture.mdx:1]: doc updated

File: src/core/VideoTexture.tsx
Comment (abernier) [inline src/core/VideoTexture.tsx:1]: simplified / dryed types and other enhancements

**Pull request merged**